### PR TITLE
SNOW-1454714 Make modin QA tests green

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1454714

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

The most change here is to change assertion for integer columns to not verify its physical length, e.g., in8 vs. int64. The physical length of integers are managed by Snowflake engine internally and can change overtime and also won't be treated as a behavior change.

### CODE REVIEW

File: tests/integ/modin/test_qcut.py [hunk 1]
```diff
@@ -278,46 +278,3 @@ def test_qcut_invalid_quantiles_negative(q):
...
     with pytest.raises(expected_exception, match=expected_message):
         snow_s.quantile(q)
-
-
-@sql_count_checker(query_count=7)
-def test_qcut_two_columns():
-    # reported by Mats Stewall, applying qcut twice leads to exploding SQL query.
-    # attempt finding a remedy
-
-    DATA_PATH = "SNOWFLAKE_SAMPLE_DATA.TPCH_SF1"
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_qcut.py:288]: remove this test because no such data in QA

File: tests/integ/modin/frame/test_fillna.py [hunk 2]
```diff
@@ -574,6 +574,9 @@ def test_df_fillna_method_reindexed_df_reordered_columns(method):
 
 
 @sql_count_checker(query_count=0)
+@pytest.mark.xfail(
+    reason="TODO SNOW-1489309 investigate why it starts to work now on qa"
```
Comment (sfc-gh-nkrishna) [inline tests/integ/modin/frame/test_fillna.py:578]: I'm assuming it doesn't work now on qa?

```suggestion
    reason="TODO SNOW-1489309 investigate failure on qa"
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
One comment, but LGTM otherwise

File: tests/integ/modin/frame/test_fillna.py (hunk 2 above)
Comment (sfc-gh-azhan) [reply tests/integ/modin/frame/test_fillna.py:578]: aha, it's a negative test so the feature works now but the test fails.

Comment (sfc-gh-joshi) [reply tests/integ/modin/frame/test_fillna.py:578]: Do you know what the new output is? Is it what we expect, or is something in the VARIANT column getting cast that wasn't before?

Comment (sfc-gh-azhan) [reply tests/integ/modin/frame/test_fillna.py:578]: Didn't go too much detail. I will investigate this diff right after.

Comment (sfc-gh-azhan) [reply tests/integ/modin/frame/test_fillna.py:578]: What I know is it does not raise exception any more.

**Review by sfc-gh-dpetersohn — APPROVED** [review]
LGTM

**Pull request merged**